src/msix/AppxFactory.cpp

```diff
@@ -95,14 +96,20 @@ namespace MSIX {
         #endif
     } CATCH_RETURN();
 
-    HRESULT STDMETHODCALLTYPE AppxFactory::CreateBundleManifestReader(IStream *inputStream, IAppxBundleManifestReader **manifestReader) noexcept
+    HRESULT STDMETHODCALLTYPE AppxFactory::CreateBundleManifestReader(IStream *inputStream, IAppxBundleManifestReader **manifestReader) noexcept try
     {
         #ifdef BUNDLE_SUPPORT
-            return static_cast<HRESULT>(Error::NotImplemented);
+            ThrowErrorIf(Error::InvalidParameter, (manifestReader == nullptr || *manifestReader != nullptr), "Invalid parameter");
+            ComPtr<IMsixFactory> self;
+            ThrowHrIfFailed(QueryInterface(UuidOfImpl<IMsixFactory>::iid, reinterpret_cast<void**>(&self)));

```

Comment:
Should not be needed: this is of type AppxFactory, which derives from IMsixFactory.  Unless there are overrides for both AppxFactory* and IMsixFactory* on AppxBundleManifestObject, it should be able to find the correct constructor and cast the type appropriately.

Reply:
can you also fix the other places where this pattern is used?

---
In reply to: [309432387](https://github.com/Microsoft/msix-packaging/pull/184#discussion_r309432387) [](ancestors = 309432387)